orderer/common/filerepo/filerepo.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package filerepo
+
+import (
+	"bufio"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type Repo struct {
+	Options Options
+}
+
+type Options struct {
+	FileSuffix          string
+	TransientFilePrefix string
+	FileRepoDir         string
+}
+
+// Save atomically persists the content to the channelName+suffix file by first writing it
+// to a tmp file prefixed by the TransientFilePrefix and then moves the file to the final
+// destination indicated by the FileSuffix.
+func (r *Repo) Save(channelName string, content []byte) error {
+	fileRepoDir := r.Options.FileRepoDir
+	dest := filepath.Join(fileRepoDir, channelName+r.Options.FileSuffix)
```

Comment:
is it possible to have the channel name a string that contains the OS file separator? If yes then maybe we need to ensure it doesn't contain such?

Reply:
I remember there are some fairly strict channel name rules, but I don't remember what characters are allowed in a channel name. I would imagine there should be some validation at the API level since you'd need to pass the channel name in via the join endpoint anyways. I do agree though that there probably could be some additional checks around the channel name and/or the file suffix and prefix

Reply:
From https://hyperledger-fabric.readthedocs.io/en/release-2.0/build_network.html#create-join-channel: 
```
Channel names must be all lower case, less than 250 characters long and match the regular expression [a-z][a-z0-9.-]*
```

Reply:
the code above which i missed, does just that.